engine/runtime-benchmarks/src/main/java/org/enso/interpreter/bench/benchmarks/semantic/FibBenchmarks.java

```diff
@@ -0,0 +1,83 @@
+package org.enso.interpreter.bench.benchmarks.semantic;
+
+import java.io.IOException;
+import java.util.Objects;
+import org.enso.common.MethodNames;
+import org.enso.compiler.benchmarks.Utils;
+import org.enso.test.utils.ContextUtils;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Value;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.infra.BenchmarkParams;
+
+/**
+ * Base class for {@link FibHostJavaPolyglotBenchmarks} and {@link FibGuestJavaPolyglotBenchmarks}.
+ */
+public abstract class FibBenchmarks {
+
+  private ContextUtils ctx;
+  private Value fib;
+
+  protected Context.Builder withModifiedContext(Context.Builder b) {
+    return b;
+  }
+
+  @Setup()
+  public void initializeBench(BenchmarkParams params) throws IOException {
+    ctx =
+        Utils.createDefaultContextBuilder().withModifiedContext(this::withModifiedContext).build();
+
+    var code =
+        """
+        polyglot java import java.lang.Math
+
+        main n=10 =
+            fib n
+
+        private fib n = if Math.max n 1 == 1 then 1 else
```

Comment:
Not so sure about the scalability of this benchmark, since this may result in Stack overflow, butt good enough for now.

Reply:
Stack was fine. The problem is the exponential time. I wasn’t patient enough to wait for fib 50 …